### PR TITLE
US 4.1 and 4.2 - Implementing Calendar Feature

### LINKED ISSUE #30: US-4.1: Connect application to Google Calendar
As a student or a guest, I want to connect my Google Calendar to the app, so that the system can access my upcoming class events and use them to help me navigate campus.

## Acceptance Criteria
- [ ] When the user clicks on the Google Calendar button, they are redirected to the Google Sign-In
- [ ] After Sign-In, the user goes back to the app and see's that the connection is successful

## Tasks
- [ ] Make a google calendar UI button
- [ ] Handle the API calls
- [ ] Handle connection between the 2 apps
- [ ] Write Test

### LINKED ISSUE #31: US-4.2: Connect multiple calendar to the application
As a logged in user, I want to view the instructions on how to properly set up my google calendar for compatability with the Campus application. 

## Acceptance Criteria
- [ ] When user selects the button 'How to set up google calendar', they can view the instructions. 
- [ ] The instructions contain screenshots and steps on how to set up the google calendar.

## Tasks
- [ ] Create UI link to view information
- [ ] Create new page for instructions

### PR DESCRIPTION
**Pull request by hibamai**
## Description:
This PR introduces the Google Calendar integration feature for the Campus App.
Users can connect their Google account, select one or more calendars, and view their class schedule directly inside the application.

The feature uses the Google Calendar API to retrieve upcoming events and displays them in a calendar view within the app. Each selected calendar is assigned a unique color to differentiate events.

Additionally, a setup guide screen was implemented to help users correctly configure their Google Calendar events so the app can extract building and room information.

The PR also includes session persistence, allowing the selected calendars and schedule state to remain available while the user stays in the app.

Finally, unit and widget tests were added to improve reliability and ensure correct behavior of models, services, and UI components.

## Related User Story:

- [(US-4.1: Connect application to Google Calendar)](https://github.com/Mini-Capstone-Fried-Chicken/Fried-Chicken/issues/29?issue=Mini-Capstone-Fried-Chicken%7CFried-Chicken%7C30)
- [(US-4.2: Connect multiple calendar to the application)](https://github.com/Mini-Capstone-Fried-Chicken/Fried-Chicken/issues/29?issue=Mini-Capstone-Fried-Chicken%7CFried-Chicken%7C31)

## Screenshot/Demo:

- Screenshot 2026-03-12 at 11 09 33 AM
- Screenshot 2026-03-12 at 11 09 54 AM
- Screenshot 2026-03-12 at 11 10 08 AM
- Screenshot 2026-03-12 at 11 10 30 AM
- Screenshot 2026-03-12 at 11 10 44 AM
- Screenshot 2026-03-12 at 11 10 58 AM
- Screenshot 2026-03-12 at 11 11 12 AM

## How to test:

1. Launch the application.
2. Navigate to the Calendar tab.
3. Click Connect to Google Calendar.
4. Sign in with a Google account (contact the dev to add your account to the accepted ones so you can connect). 
5. Grant calendar access permissions.
6. Select one or more calendars that contain class events.
7. Click Continue.
8. Verify that events appear in the calendar schedule view.
9. Switch between Day / Week / Month / Schedule views.
10. Press How to set up calendar to verify the setup guide screen.
11. Press the back button from the schedule view to return to calendar selection.
12. Logout from Settings and verify that the Google session disconnects.

## Type of Change:
- [x] Feature
- [ ] Bug fix
- [ ] Refactor
- [ ] Documentation update
- [x] Tests
- [ ] Others

## Pull Request Checklist
- [x] Flutter CI test passed
- [x] Sonar test passed
- [x] Codecov test passed
- [ ] Application runs without errors
- [ ] Feature tested locally on a high level
- [ ] Architecture conventions respected
- [ ] No API keys or secrets committed
- [ ] Document when necessary

## PR Review Checklist (dev)
- [ ] Code is readable and well structured
- [ ] No obvious bugs or edge cases missed
- [ ] Tests are adequate
- [ ] Naming and organization are clear
- [ ] Feature works as described
- [ ] No unnecessary complexity
- [ ] Follows project architecture

## PR Review Checklist (QA)
- [ ] Made sure to test the application fully
- [ ] No bugs created by this PR

Closes #30 
Closes #31

### CODE REVIEW

File: lib/features/calendar/data/models/google_calendar_event.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  const GoogleCalendarEvent({
+    required this.id,
+    required this.title,
+    required this.start,
+    required this.end,
+    required this.location,
+    required this.calendarId,
+    required this.calendarName,
+    required this.color,
+  });
```
Comment (enyc24) [inline lib/features/calendar/data/models/google_calendar_event.dart]: This function has a lot of parameters. Is there any way you can reduce the number of parameters using helper methods?

File: lib/features/calendar/ui/google_calendar_setup_screen.dart [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                ),
+              ),
+              const SizedBox(height: 32),
+
+              _StepSection(
+                title: 'Step 1',
+                description:
+                    'Click on the + sign at the bottom right of your screen and choose event.',
+                imagePath: 'assets/images/google_calendar_step1.png',
+              ),
```
Comment (enyc24) [inline lib/features/calendar/ui/google_calendar_setup_screen.dart]: Use a constant with this constructor to improve performance

File: lib/features/calendar/data/models/google_calendar_event.dart (hunk 1 above)
Comment (hibamai) [reply lib/features/calendar/data/models/google_calendar_event.dart]: Good suggestion. However, since this is a simple data model and the parameters map directly to the event fields, I kept the explicit named parameters for clarity when constructing instances.

File: lib/features/calendar/ui/google_calendar_setup_screen.dart (hunk 2 above)
Comment (hibamai) [reply lib/features/calendar/ui/google_calendar_setup_screen.dart]: done

File: lib/features/calendar/ui/google_calendar_setup_screen.dart [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+              const SizedBox(height: 28),
+
+              _StepSection(
+                title: 'Step 2',
+                description:
+                    'Set the location of the event as the building name/code your class is in and the description as the room number.',
+                imagePath: 'assets/images/google_calendar_step2.png',
+              ),
```
[the remark this answers is not in the document]
Comment (hibamai) [reply lib/features/calendar/ui/google_calendar_setup_screen.dart]: done

File: lib/features/calendar/ui/widgets/calendar_schedule_view.dart [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                    style: TextStyle(fontSize: 15, color: Colors.black54),
+                  ),
+                )
+              : SfCalendar(
+                  key: ValueKey(_calendarView),
+                  view: _calendarView,
+                  dataSource: dataSource,
+                  initialDisplayDate: widget.events.isNotEmpty
+                      ? widget.events.first.start
+                      : DateTime.now(),
```
[the remark this answers is not in the document]
Comment (hibamai) [reply lib/features/calendar/ui/widgets/calendar_schedule_view.dart]: done
